src/main/java/hudson/model/ExternalJob.java

```diff
@@ -76,7 +78,11 @@ public synchronized ExternalRun newBuild() throws IOException {
         long timeSinceLast = System.currentTimeMillis() - lastBuildStartTime;
         if (timeSinceLast < 1000) {
             try {
-                Thread.sleep(1000 - timeSinceLast);
+                while(timeSinceLast > 0){
+                    wait();
```

Comment:
@kwhetstone better to bump baseline to 1.603.3 and remove the whole loop: build dirs are not longer timestamp-based a that baseline.